src/test/scala/it/pagopa/interop/catalogmanagement/model/persistence/serializer/PersistentDeserializationSpec.scala

```diff
@@ -0,0 +1,312 @@
+import cats.implicits._
+import com.softwaremill.diffx.Diff
+import com.softwaremill.diffx.generic.auto._
+import com.softwaremill.diffx.munit.DiffxAssertions
+import it.pagopa.interop.catalogmanagement.model._
+import it.pagopa.interop.catalogmanagement.model.persistence._
+import it.pagopa.interop.catalogmanagement.model.persistence.serializer.PersistentSerializationSpec._
+import it.pagopa.interop.catalogmanagement.model.persistence.serializer.v1.catalog_item.AgreementApprovalPolicyV1.{
+  AUTOMATIC,
+  MANUAL
+}
+import it.pagopa.interop.catalogmanagement.model.persistence.serializer.v1.catalog_item.CatalogDescriptorStateV1._
+import it.pagopa.interop.catalogmanagement.model.persistence.serializer.v1.catalog_item._
+import it.pagopa.interop.catalogmanagement.model.persistence.serializer.v1.events._
+import it.pagopa.interop.catalogmanagement.model.persistence.serializer.v1.state._
+import munit.ScalaCheckSuite
+import org.scalacheck.Gen
+import org.scalacheck.Prop.forAll
+
+import java.time.format.DateTimeFormatter
+import java.time.{OffsetDateTime, ZoneOffset}
+import scala.reflect.runtime.universe.{TypeTag, typeOf}
+
+class PersistentDeserializationSpec extends ScalaCheckSuite with DiffxAssertions {
+
+  deserCheck[State, StateV1](stateGen)
+  deserCheck[CatalogItemAdded, CatalogItemV1AddedV1](catalogItemAddedGen)
+  deserCheck[ClonedCatalogItemAdded, ClonedCatalogItemV1AddedV1](clonedCatalogItemAddedGen)
+  deserCheck[CatalogItemUpdated, CatalogItemV1UpdatedV1](catalogItemUpdatedGen)
+  deserCheck[CatalogItemWithDescriptorsDeleted, CatalogItemWithDescriptorsDeletedV1](
+    catalogItemWithDescriptorsDeletedGen
+  )
+  deserCheck[CatalogItemDeleted, CatalogItemDeletedV1](catalogItemDeletedGen)
+  deserCheck[CatalogItemDocumentDeleted, CatalogItemDocumentDeletedV1](catalogItemDocumentDeletedGen)
+  deserCheck[CatalogItemDescriptorAdded, CatalogItemDescriptorAddedV1](catalogItemDescriptorAddedGen)
+  deserCheck[CatalogItemDescriptorUpdated, CatalogItemDescriptorUpdatedV1](catalogItemDescriptorUpdatedGen)
+  deserCheck[CatalogItemDocumentAdded, CatalogItemDocumentAddedV1](catalogItemDocumentAddedGen)
+  deserCheck[CatalogItemDocumentUpdated, CatalogItemDocumentUpdatedV1](catalogItemDocumentUpdatedGen)
+  deserCheck[MovedAttributesFromEserviceToDescriptors, MovedAttributesFromEserviceToDescriptorsV1](movedGen)
+
+  // TODO move me in commons
+  def deserCheck[A, B: TypeTag](
+    gen: Gen[(A, B)]
+  )(implicit e: PersistEventDeserializer[B, A], loc: munit.Location, d: => Diff[Either[Throwable, A]]): Unit =
+    property(s"${typeOf[B].typeSymbol.name.toString} is correctly deserialized") {
+      forAll(gen) { case (state, stateV1) =>
+        // * This is declared lazy in the signature to avoid a MethodTooBigException
+        implicit val diffX: Diff[Either[Throwable, A]] = d
+        assertEqual(PersistEventDeserializer.from[B, A](stateV1), Right(state))
+      }
+    }
+}
+
+object PersistentDeserializationSpec {
+
```

Comment:
let's add a comment to remind us why this has been duplicated, like
```suggestion
// Generators have been duplicated because the conversion is not isomorphic for attributes
```